tf-psa-crypto/drivers/builtin/src/aesni.c

```diff
@@ -489,7 +489,7 @@ int mbedtls_aesni_crypt_ecb(mbedtls_aes_context *ctx,
          "movdqu    %%xmm0, (%4)    \n\t" // export output
          :
          : "r" (ctx->nr), "r" (ctx->buf + ctx->rk_offset), "r" (mode), "r" (input), "r" (output)
-         : "memory", "cc", "xmm0", "xmm1");
+         : "memory", "cc", "xmm0", "xmm1", "0", "1");
 
```

Comment:
I'm checking the other asm blocks in this file.

In `aesni_setkey_enc_128`, I think `xmm0` and `xmm1` are missing from the clobber list, right? And in `aesni_setkey_enc_192` and `aesni_setkey_enc_256`, same and also `xmm2`. With your fix, the rest look fine to me.

Reply:
> In `aesni_setkey_enc_128`, I think `xmm0` and `xmm1` are missing from the clobber list, right? And in `aesni_setkey_enc_192` and `aesni_setkey_enc_256`, same and also `xmm2`. With your fix, the rest look fine to me.

Oh, you're right. Is this something you'd fix separately from this PR?

OTOH, I think in `mbedtls_aesni_gcm_mult`, the clobber list unnecessarily includes `cc` - I think none of those instructions modify flags. But this may be better to leave as-is at this point.

Reply:
I'd appreciate it if you could fix those clobber lists while you're at it, so we can say we fixed the assembly in the AESNI code and not just in one function. But if not we'll make a follow-up pull request.

Reply:
OK. I intend to fix these as well (and credit you in the commit message for noticing them), but it's a busy week and it's taking me a while to get back to "free time" work again. Just letting you know that I accepted the task, but can't handle it as a high priority.

Reply:
I've just added a commit fixing the missed XMM clobbers and adding a changelog entry. I hope this is as desired.

I tested these same changes in a copy/revision of the code as we integrated it in JtR. I temporarily reverted from usage of intrinsics to asm, rebuilt JtR, and ran our tests - so whatever functions we do use there were tested. I also checked the `aesni.o` text section size, which remained unchanged at 1445 bytes for me (which does not guarantee no increase in actually used code as the function entry points are aligned, so each previous function is padded).

I never tried building/testing mbedTLS proper (sorry!) and quickly trying to do so now first gave me this:

```
$ make check
Makefile:19: *** /framework/exported.make not found (and does not appear to be a git checkout). Please ensure you have downloaded the right archive from the release page on GitHub..  Stop.
```

which made me look inside the `Makefile`, see the other error message nearby (is the condition on which message to print maybe wrong?) and then do:

```
$ git submodule update --init
Submodule 'framework' (https://github.com/Mbed-TLS/mbedtls-framework) registered for path 'framework'
Cloning into '/home/user/mbedtls/framework'...
Submodule path 'framework': checked out 'df0144c4a3c0fc9beea606afde07cf8708233675'
```

Then rerunning `make check` fails at:

```
  CC    src/test_helpers/ssl_helpers.c
make[1]: Leaving directory '/home/user/mbedtls/tests'
make[1]: Entering directory '/home/user/mbedtls/library'
  Gen   ../tf-psa-crypto/core/psa_crypto_driver_wrappers.h ../tf-psa-crypto/core/psa_crypto_driver_wrappers_no_static.c
Traceback (most recent call last):
  File "/home/user/mbedtls/library/../scripts/generate_driver_wrappers.py", line 18, in <module>
    import jsonschema
ModuleNotFoundError: No module named 'jsonschema'
```

I'm not eager to install a Python module without creating a dedicated environment for this testing first, so I stopped here.

I mention this as maybe-useful feedback on maybe improving error messages and maybe relaxing build/test dependencies for new users.

Reply:
> OK. I intend to fix these as well (and credit you in the commit message for noticing them), but it's a busy week and it's taking me a while to get back to "free time" work again. Just letting you know that I accepted the task, but can't handle it as a high priority.

No worries, I know all about “free time”! If you prefer, I can take over and finish the patch. Well, you've gone ahead and updated, thank you very much, but I can take over if there further updates are needed.

> I never tried building/testing mbedTLS proper (sorry!) and quickly trying to do so now first gave me this: (…)

Feedback noted. Unfortunately, while we'd like to get rid of the Python dependencies, that would require significant engineering work. For what it's worth, releases should be fine on both counts, you just run into these difficulties when you download from a development branch. Hey, at least you didn't have to run autotools!